library/std/src/sync/mpmc/list.rs

```diff
@@ -564,7 +566,7 @@ impl<T> Channel<T> {
             // In that case, just wait until it gets initialized.
             while block.is_null() {
                 backoff.spin_heavy();
-                block = self.head.block.load(Ordering::Acquire);
+                block = self.head.block.swap(ptr::null_mut(), Ordering::AcqRel);
```

Comment:
Might be worth a comment explaining why this has to be `swap`?

Reply:
Added a comment explaining the invariant that needs to be maintained